pkg/controller/machine/machine_controller.go

```diff
@@ -245,6 +230,29 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 	return reconcile.Result{}, nil
 }
 
+// needUpdateFinalizers appends any missing finalizers to the machine
+// and returns true if the api server needs to be updated.
+func needUpdateFinalizers(m *clusterv1.Machine, cluster *clusterv1.Cluster) bool {
```

Comment:
Should this be called `updateFinalizers` instead? I wouldn't expect a method called `needUpdateFinalizers` to update.